src/main/java/io/usethesource/vallang/IList.java

```diff
@@ -22,6 +22,12 @@
 import io.usethesource.vallang.visitors.IValueVisitor;
 
 public interface IList extends ICollection<IList> {
+    
+    @Override
+    default int getMatchFingerprint() {
+        return 3322014; // "list".hashCode()
```

Comment:
I'm a bit confused why `IConstructor` is the only one where the name flows into the fingerpint (and arity for tuples)? Shouldn't we also let the type of the list flow into the fingerprint?

Reply:
Good question. No, we should always avoid letting the type of the list flow into the fingerprint. The reason is that patterns with either less specific types, or more specific types, still have to match. The only requirement for the case to match in general is _type comparability_. And so we use only the _kind_ of the type to determine the fingerprint, and not the actual type. 

More generally, the fingerprint contract is against the pattern-matching semantics: the contract is that if a pattern would match, then the fingerprint of the pattern should be equal to the fingerprint of the subject.

For constructors type kinds are very specific: the name of the constructor, the name of the data-type, etc, are all fine to hash on, since the pattern will match if-and-only-if those names are equal. The same holds for INode instances. However, for lists all we basically can switch on is that it is a list. The recursive pattern match, will then make sure that the elements are of a certain type, etc, following the structure of the pattern as it becomes more specific.

Reply:
ah true, thanks, it's a very specific fingerprint.